apps/api/src/persistence/repositories/connectionRepository.e2e.spec.ts

```diff
@@ -0,0 +1,178 @@
+
+        databaseProvider = new DatabaseProvider(arangoConnectionProvider);
+
+        testRepositoryProvider = new TestRepositoryProvider(databaseProvider);
+
+        app = moduleRef.createNestApplication();
+
+        await app.init();
+    });
+
+    afterAll(async () => {
+        await testRepositoryProvider.testTeardown();
+
+        await app.close();
+    });
+
+    const { connections, resources } = testData;
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+
+        await testRepositoryProvider.addEntitiesOfManyTypes(resources);
+
+        await testRepositoryProvider.getEdgeConnectionRepository().createMany(connections);
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
+
+    describe('fetchMany', () => {
+        it('should return all edge connections', async () => {
+            const fetchManyResult = await testRepositoryProvider
+                .getEdgeConnectionRepository()
+                .fetchMany();
+
+            expect(fetchManyResult).toEqual(connections);
+        });
+    });
+
+    describe('getCount', () => {
+        it('should return the correct number of edge connections', async () => {
+            const count = await testRepositoryProvider.getEdgeConnectionRepository().getCount();
+
+            const expectedCount = connections.length;
+
+            expect(count).toBe(expectedCount);
+        });
+    });
+
+    describe('fetchById', () => {
+        describe('when there is no edge connection with the given id', () => {
+            it('should return NotFound', async () => {
+                const result = await testRepositoryProvider
+                    .getEdgeConnectionRepository()
+                    .fetchById('bogus-id-193949');
+
+                expect(result).toBe(NotFound);
+            });
+        });
+
+        connections.forEach((connection) =>
+            describe(`when there is an edge connection with the given id: ${connection.id}`, () => {
+                it('should return the entity', async () => {
+                    const expectedResult = connections.find(
+                        ({ id: testInstanceId }) => testInstanceId === connection.id
+                    );
+
+                    const actualResult = await testRepositoryProvider
+                        .getEdgeConnectionRepository()
+                        .fetchById(connection.id);
+
+                    // In case expectedResult didn't find anything with the search
+                    expect(actualResult).toBeTruthy();
+
+                    expect(actualResult).toEqual(expectedResult);
+                });
+            })
+        );
+    });
+
+    describe('create', () => {
+        it('should create a new edge connection', async () => {
+            const uniqueNewId = 'brand-new-id-123';
+
+            const edgeConnectionToCreate = connections[0].clone({ id: uniqueNewId });
```

Comment:
Would we not want to complain if there was a duplicate edge saved to the database where all the _to, _from, and members, etc. are the same?

Reply:
That's a good point. That should be part of the `external state` validation for an edge connection. Just as for a resource, we have to make sure there is no other resource with the same ID (which Arango conveniently prevents, but we should trust but verify there at very least to avoid a leaky db abstraction), we need to do this. 

At the same time, this is a rare edge case and I think we can probably add this safeguard if \ when we hit the problem. Note that each edge document has a unique _key that maps to a unique id for the edge connection, so the place to prevent this duplication is really when we add the commands to create \ update edge connections.

Reply:
Right, I'm just thinking of the scenario in a rudimentary web form where you hit submit twice and generate two new records in the db with distinct ids but the same information from the form fields.